src/noit_check_tools.h

```diff
@@ -68,7 +68,7 @@ API_EXPORT(void)
   noit_check_make_attrs(noit_check_t *check, mtev_hash_table *attrs);
 
 #define CHOOSE_EVENTER_THREAD_FOR_CHECK(check) \
-  eventer_choose_owner(((int)((*((intptr_t *)check->checkid))/sizeof(*(check))))*2654435761)
+  eventer_choose_owner(((int)((*((unsigned char *)check->checkid))/sizeof(*(check))))*2654435761)
```

Comment:
the int cast is in the wrong place.  casting to `intptr_t` violates type safety and won't compile in strict mode.